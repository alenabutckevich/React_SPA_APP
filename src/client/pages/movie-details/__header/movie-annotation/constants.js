export const BASE_CLASS = 'movie_annotation';