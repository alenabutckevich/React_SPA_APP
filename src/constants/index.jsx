export const START_FETCHING_MOVIES_BY_TITLE = 'START_FETCHING_MOVIES';
export const FETCH_MOVIES_BY_TITLE_SUCCESS = 'FETCH_MOVIES_SUCCESS';
export const FETCH_MOVIES_BY_TITLE_FAILURE = 'FETCH_MOVIES_FAILURE';
export const START_FETCHING_MOVIES_BY_DIRECTOR = 'START_FETCHING_MOVIES';
export const FETCH_MOVIES_BY_DIRECTOR_SUCCESS = 'FETCH_MOVIES_SUCCESS';
export const FETCH_MOVIES_BY_DIRECTOR_FAILURE = 'FETCH_MOVIES_FAILURE';
export const START_FETCHING_MOVIE_BY_ID = 'START_FETCHING_MOVIE_BY_ID';
export const FETCH_MOVIE_BY_ID_SUCCESS = 'FETCH_MOVIE_BY_ID_SUCCESS';
export const FETCH_MOVIE_BY_ID_FAILURE = 'FETCH_MOVIE_BY_ID_FAILURE';
export const START_FETCHING_MOVIE_CAST_AND_CREW = 'START_FETCHING_MOVIE_CAST_AND_CREW';
export const FETCH_MOVIE_CAST_AND_CREW_SUCCESS = 'FETCH_MOVIE_CAST_AND_CREW_SUCCESS';
export const FETCH_MOVIE_CAST_AND_CREW_FAILURE = 'FETCH_MOVIE_CAST_AND_CREW_FAILURE';
export const SET_SORT_FILTER = 'SET_SORT_FILTER';
export const SET_SEARCH_FILTER = 'SET_SEARCH_FILTER';
export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY';
export const RESET_CURRENT_MOVIE = 'RESET_CURRENT_MOVIE';
export const START_FETCHING_GENRES = 'START_FETCHING_GENRES';
export const FETCH_GENRES_SUCCESS = 'FETCH_GENRES_SUCCESS';
export const FETCH_GENRES_FAILURE = 'FETCH_GENRES_FAILURE';
