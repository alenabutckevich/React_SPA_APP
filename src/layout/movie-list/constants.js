export const RELEASE_DATE_FILTER = {
    name: 'release date',
    value: 'release_date'
}

export const RATING_FILTER = {
    name: 'rating',
    value: 'vote_average'
}

export const TITLE_FILTER = {
    name: 'title',
    value: 'title'
}