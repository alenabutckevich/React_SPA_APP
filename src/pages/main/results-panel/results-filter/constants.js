export const RELEASE_DATE_FILTER = 'release date';
export const RATING_FILTER = 'rating';