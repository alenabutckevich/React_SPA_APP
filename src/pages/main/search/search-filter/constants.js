export const TITLE_FILTER = 'title';
export const DIRECTOR_FILTER = 'director';