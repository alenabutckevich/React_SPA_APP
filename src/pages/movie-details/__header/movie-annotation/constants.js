export const BASE_CLASS = 'movie-annotation';