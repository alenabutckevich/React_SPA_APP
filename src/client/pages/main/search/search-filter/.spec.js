import React from 'react';
import toJson from 'enzyme-to-json';
import SearchFilter from './search-filter';
import { TITLE_FILTER, DIRECTOR_FILTER } from './constants';

const setup = propOverrides => {
    const { currentFilter } = propOverrides;
    const props = Object.assign({
        currentFilter: currentFilter,
        changeFilter: jest.fn()
    }, propOverrides);

    const wrapper = shallow(<SearchFilter {...props}/>);

    return {
        wrapper,
        props
    }
}

describe('<SearchFilter />', () => {

    it('render search filter - director', () => {
        const { wrapper, props: {changeFilter} } = setup({currentFilter: DIRECTOR_FILTER});   
        const link = wrapper.find('.search_filter__button_active');

        expect(link.text()).toEqual(DIRECTOR_FILTER);
        link.simulate('click');
        expect(changeFilter).toBeCalled();

        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('render search filter - title', () => {
        const { wrapper, props: {changeFilter} } = setup({currentFilter: TITLE_FILTER});
        const link = wrapper.find('.search_filter__button_active');

        expect(link.text()).toEqual(TITLE_FILTER);
        link.simulate('click');
        expect(changeFilter).toBeCalled();
       
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})