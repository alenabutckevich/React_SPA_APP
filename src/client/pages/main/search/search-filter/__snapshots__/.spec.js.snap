// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFilter /> render search filter - director 1`] = `
<div
  className="search_filter"
>
  <span
    className="search_filter__header"
  >
    search by
  </span>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <button
      className="search_filter__button "
      key="1"
      onClick={[Function]}
    >
      title
    </button>
    <button
      className="search_filter__button search_filter__button_active"
      key="2"
      onClick={[Function]}
    >
      director
    </button>
  </div>
</div>
`;

exports[`<SearchFilter /> render search filter - title 1`] = `
<div
  className="search_filter"
>
  <span
    className="search_filter__header"
  >
    search by
  </span>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <button
      className="search_filter__button search_filter__button_active"
      key="1"
      onClick={[Function]}
    >
      title
    </button>
    <button
      className="search_filter__button "
      key="2"
      onClick={[Function]}
    >
      director
    </button>
  </div>
</div>
`;
